cli/cmd/test_connection.go

```diff
@@ -72,22 +72,42 @@ func updateSyncTestConnectionStatus(ctx context.Context, logger zerolog.Logger,
 	}
 
 	log.Info().Str("status", string(status)).Msg("Sending sync test connection to API")
-	requestBody := cloudquery_api.UpdateSyncTestConnectionJSONRequestBody{
-		Status: status,
-	}
-	if failedTestResult != nil {
-		requestBody.FailureCode = &failedTestResult.FailureCode
-		requestBody.FailureReason = &failedTestResult.FailureDescription
-	}
-	res, err := apiClient.UpdateSyncTestConnectionWithResponse(ctx, teamName, syncTestConnectionUUID, requestBody)
-	if err != nil {
-		log.Warn().Err(err).Msg("Failed to send sync test connection to API")
-		return
+
+	var statusCode int
+	switch os.Getenv("_CQ_SYNC_TEST_CONNECTION_KIND") {
+	case "source":
+		requestBody := cloudquery_api.UpdateSyncTestConnectionForSyncSourceJSONRequestBody{
+			Status: status,
+		}
+		if failedTestResult != nil {
+			requestBody.FailureCode = &failedTestResult.FailureCode
+			requestBody.FailureReason = &failedTestResult.FailureDescription
+		}
+		res, err := apiClient.UpdateSyncTestConnectionForSyncSourceWithResponse(ctx, teamName, syncTestConnectionUUID, requestBody)
+		if err != nil {
+			log.Warn().Err(err).Msg("Failed to send sync test connection result to API")
+			return
+		}
+		statusCode = res.StatusCode()
+	default:
```

Comment:
Maybe the default should be to do nothing? And on destination it should update the destination?
That way if we add other kinds (e.g. transformer) this won't break

Reply:
Handled in https://github.com/cloudquery/cloudquery/pull/18790